### DISCUSSION TITLE
Maintain vulnerabilities of components used by DT?

### DISCUSSION THREAD

**Discussion by jbruenner** [DependencyTrack/dependency-track#1764] [Q&A]
Hi,

our Kubernetes infrastructure is scanned by Trivy. It reports a lot of vulnerabilities of the operating 
system used by DT. I'm asked to accept the risk of this vulnerabilities. I can't answer this for sure: I 
don't know what theses components are used for. So I can't tell good from bad :-(

Do you maintain a complete list of known vulnerabilities, telling which one is acceptable and where
I have to set up a work around?

Thanks a lot
Jörg

**Comment by nscuro** [comment]
Hi @jbruenner, I understand this is an issue. As of now, we don't have such a document.

Ultimately we'd like to publish a [VEX document](https://cyclonedx.org/use-cases/#vulnerability-exploitability) for DT, which you as DT user could then ingest into your DT instance again. Essentially dogfooding our own feature. Of course you could also inspect it manually if need be.

That being said, scanner results will differ from tool to tool, so I can't guarantee that the VEX we'll eventually publish will cover all vulnerabilities found by Trivy. There will also most likely be a delay between your scanner identifying new issues and us updating our VEX document.

Incidentally, we use Trivy to scan our container as well. Results are published to GitHub's security scanning feature, where we then can dismiss alerts:

![image](https://user-images.githubusercontent.com/5693141/177603132-6e631779-605c-464b-a7ce-08989984caad.png)

However, only us maintainers see it, and GitHub doesn't allow for a detailed analysis to be provided. There's also no way to export this sadly.

Looking at the current issues found by Trivy, I can provide you the following feedback:

* CVE-2022-2068 affects the `c_rehash` script of OpenSSL, which to my understanding would need to be explicitly executed. DT does not do this. To exploit this, you'd have to manually execute that script inside the container, on a directory where an attacker managed to place a file with a name that contains shell commands.
* CVE-2022-1271 affects the `zgrep` CLI tool. Again something that'd need to be manually executed with an attacker-supplied input. Highly unlikely.
* CVE-2022-34903 is an informative finding and not applicable either.

If there are more vulnerabilities that cause you trouble right now, let me know and I can provide a quick response that you can use as justification.

**Comment by jbruenner** [comment]
Hi @nscuro,

thank you.

> Ultimately we'd like to publish a VEX document for DT,

Great, it looks like the solution of my problems :-) When do you plan to start it?

> If there are more vulnerabilities that cause you trouble right now, let me know and I can provide a quick response that you can use as justification.

Thank you :-) Here is the complete list of critical and high ratet vulnerabilities. I marked the ones I got already a statement from you.

**frontend**
Vulnerability		Severity	CVSS3		Package		installed version	fixed version	no risk
CVE-2022-27404	Critical	nvd: 9.8		freetype		2.11.1-r0			2.11.1-r1
CVE-2022-1586	Critical	nvd: 9.1		pcre2		10.39-r0			10.40-r0
CVE-2022-1587	Critical	nvd: 9.1		pcre2		10.39-r0			10.40-r0
CVE-2022-22576	High	redhat: 4.6	curl			7.80.0-r0			7.80.0-r1
CVE-2022-27775	High	redhat: 2.6	curl			7.80.0-r0			7.80.0-r1
CVE-2022-22576	High	redhat: 4.6	libcurl		7.80.0-r0			7.80.0-r1
CVE-2022-27775	High	redhat: 2.6	libcurl		7.80.0-r0			7.80.0-r1
CVE-2022-27405	High	nvd: 7.5		freetype		2.11.1-r0			2.11.1-r2
CVE-2022-27406	High	nvd: 7.5		freetype		2.11.1-r0			2.11.1-r2
CVE-2022-27780	High	nvd: 7.5		curl			7.80.0-r0			7.80.0-r2
CVE-2022-27781	High	nvd: 7.5		curl			7.80.0-r0			7.80.0-r2
CVE-2022-27782	High	nvd: 7.5		curl			7.80.0-r0			7.80.0-r2
CVE-2022-27780	High	nvd: 7.5		libcurl		7.80.0-r0			7.80.0-r2
CVE-2022-27781	High	nvd: 7.5		libcurl		7.80.0-r0			7.80.0-r2
CVE-2022-27782	High	nvd: 7.5		libcurl		7.80.0-r0			7.80.0-r2

**apiserver**
Vulnerability		Severity	CVSS3	Package	installed version	fixed version		no risk
CVE-2022-1292	Critical	nvd: 9.8	libssl1.1	1.1.1n-0+deb11u1	1.1.1n-0+deb11u2	x
CVE-2022-1664	Critical			dpkg	1.20.9			1.20.10			x
CVE-2021-23463	Critical	nvd: 9.1	h2		1.4.200			2.0.202			x
CVE-2021-42392	Critical	nvd: 9.8	h2		1.4.200			2.0.206			x
CVE-2022-23221	Critical	nvd: 9.8	h2		1.4.200			2.1.210			x
CVE-2022-2068	Critical	nvd: 9.8	libssl1.1	1.1.1n-0+deb11u1	1.1.1n-0+deb11u3	x

@all: 
Am I the only one with this kind of trouble? How do you sail around this rocks in restricted environments
without VEX?

Thank you for your help
Jörg

**Reply by zgael** [reply]
Hi there, 
@jbruenner I can assure you you're not the only one with this kind of trouble!
Quick summary of what's going on by my side : we need to prove the images we're putting inside our environments are clean of unfixed critical CVEs - so quite similar to you.
Also using trivy, and we have the same scan results (using `--ignore-unfixed` and the latest versions as of today - apiserver:4.5.0 and frontend:4.5.1).

However, you said you marked the CVEs that were already discussed, but I can only find CVE-2022-1271, CVE-2022-2068, CVE-2022-34903.

@jbruenner Is there some other channel where you exchanged CVE justifications ? 

So here is the list I came up with, where I believe many CVEs remain unexplained (at least on this thread) : 

**frontend : **
- CVE-2022-27404 - freetype
- CVE-2022-1586 - pcre2
- CVE-2022-1587 - pcre2

**apiserver : **
- CVE-2022-1664 - dpkg
- CVE-2022-2068 - libssl1.1 -> directly justified
- CVE-2022-1292 - libssl1.1 -> indirectly justified, same than previous one
- CVE-2021-23463 - h2
- CVE-2021-42392 - h2
- CVE-2022-23221 - h2

Don't hesitate to ask if you need some more precisions, and thanks for your help!

Gaël

**Comment by jbruenner** [comment]
Hi all,

@zgael: I opened some weeks ago  #1740. There I got the answers concerning the apiserver. 
I don't have any information concerning frontend vulnerabilities:

vulnerability            severity               package   inst. vers.    latest version
CVE-2022-27404   Critical nvd: 9.8    freetype  2.11.1-r0    2.11.1-r1
CVE-2022-1586     Critical nvd: 9.1    pcre2     10.39-r0     10.40-r0
CVE-2022-1587     Critical nvd: 9.1    pcre2     10.39-r0     10.40-r0
CVE-2022-22576   High redhat: 4.6   curl        7.80.0-r0    7.80.0-r1
CVE-2022-27775   High redhat: 2.6   curl        7.80.0-r0    7.80.0-r1
CVE-2022-22576   High redhat: 4.6   libcurl     7.80.0-r0    7.80.0-r1
CVE-2022-27775   High redhat: 2.6   libcurl     7.80.0-r0    7.80.0-r1
CVE-2022-27405   High nvd: 7.5       freetype  2.11.1-r0    2.11.1-r2
CVE-2022-27406   High nvd: 7.5       freetype  2.11.1-r0    2.11.1-r2
CVE-2022-27780   High nvd: 7.5       curl         7.80.0-r0    7.80.0-r2
CVE-2022-27781   High nvd: 7.5       curl         7.80.0-r0    7.80.0-r2
CVE-2022-27782   High nvd: 7.5       curl         7.80.0-r0    7.80.0-r2
CVE-2022-27780   High nvd: 7.5       libcurl     7.80.0-r0    7.80.0-r2
CVE-2022-27781   High nvd: 7.5       libcurl     7.80.0-r0    7.80.0-r2
CVE-2022-27782   High nvd: 7.5       libcurl     7.80.0-r0    7.80.0-r2

Thank you for your help
Jörg
